### ISSUE TITLE
AddressParser: add provider lookup methods

### ISSUE THREAD

**Issue by msgilligan** [bitcoinj/bitcoinj#3022]
The `AddressParser` interface should have at least one provider look up method (that work with JDK 8 providers and in the future will work with JDK 9+ providers)  Some possibilities:

1. `AddressParser.get()`: gets a "default" parser that is configured to parse all known/configured variants
2. `AddressParser.for(Network network)`: gets a validating parser that will only successfully parse addresses for the given network.
3. `NetworkAddressParser.get()` returns a parser than validate for all known networks and for a specified network (see suggestions in Issue #3021)

**Comment by schildbach** [comment]
I think this can be deferred to 0.18.

**Comment by msgilligan** [comment]
> I think this can be deferred to 0.18.

Agreed. Note that dead2c74ab962215aed6f5ecd52f0678fd02ff65 added `getDefault()` and `getDefault(Network)` which could be updated to use `ServiceLocator` to find default implementations.
